### DISCUSSION TITLE
Calculating  VIMP CI for conditional forests

### DISCUSSION THREAD

**Discussion by chanaka88** [kogalur/randomForestSRC#219] [Q&A] [answered]
I'm impressed by reading your work to calculate standard errors and confidence intervals for variable importance in random forests. I wonder whether  I can fit a model with highly correlated features using rfsrc.   I  have used cforest in r to estimate VIMP for conditional forests; however, the package does not offer a solution to calculate confidence intervals.  
 1. Can rfsrc be applied to highly correlated features?
2. Can subsampling be used to calculate confident intervals in conditional random forests VIMP.

**Comment by ishwaran** [comment, marked as answer]
There's a two part answer to your questions.  First, the subsampling method **does** apply to correlated features.  Under the assumptions for the subsampling method, the confidence interval calculated does what it's supposed to: namely it provides coverage around the true VIMP parameter.  That's the first part.  For the second part of the answer, we have to recognize that all current variable importance methods have difficulties in the correlated feature setting, especially if there is high correlation.  Therefore, there is no guarantee that the confidence interval will hover around zero for weak variables and away from zero for strong variables, because there is no guarantee that the true VIMP parameter is equal to the true signal value.  It should work reasonably well, but the reality is there are current limitations to what variable importance can do in the correlated features problem.  Hopefully we will have some solutions to this in the future but in the meantime you can certainly use the function for your analysis.  Finally, the function that we provide is only designed to work with our package and random forests and not conditional random forests.
